pandas/formats/format.py

```diff
@@ -1951,7 +1951,12 @@ def _format_hierarchical_rows(self):
                 for spans, levels, labels in zip(level_lengths,
                                                  self.df.index.levels,
                                                  self.df.index.labels):
-                    values = levels.take(labels)
+
+                    if levels._can_hold_na:
+                        values = levels.take(labels, fill_value=True)
```

Comment:
you don't need the conditional - just pass fill_valiue
it won't have an effect of no nas